src/Joystick/JoystickAndroid.h

```diff
@@ -0,0 +1,49 @@
+#ifndef JOYSTICKANDROID_H
+#define JOYSTICKANDROID_H
+
+#include "Joystick.h"
+#include "Vehicle.h"
+#include "MultiVehicleManager.h"
+
+#include <jni.h>
+#include <QtCore/private/qjni_p.h>
+#include <QtCore/private/qjnihelpers_p.h>
+#include <QtAndroidExtras/QtAndroidExtras>
+#include <QtAndroidExtras/QAndroidJniObject>
+
+
+class JoystickAndroid : public Joystick, public QtAndroidPrivate::GenericMotionEventListener, public QtAndroidPrivate::KeyEventListener
+{
+public:
+    JoystickAndroid(const QString& name, int axisCount, int buttonCount, int id, MultiVehicleManager* multiVehicleManager);
+    ~JoystickAndroid();
+
+    static QMap<QString, Joystick*> discover(MultiVehicleManager* _multiVehicleManager); 
+
+private:
+    bool handleKeyEvent(jobject event);
```

Comment:
Coding style is _fnname and _membername for all private/protected function and member names.